Fixed: Search results disappear after typing a space.

MohitMaliFtechiz opened the pull request:
Fixes #4456 

* This issue occurred because the exact text from the search box was passed, including unnecessary spaces, which caused libkiwix to return no results. The search term is now trimmed before being passed to libkiwix, ensuring extra spaces are removed.
* The same fix has been applied to PageFragment (for bookmarks, notes, and history) as well as the language list and online content search.
* Added UI test case for this so that we can avoid this type of errors in future.

| Before Fix | After Fix |
|---------------|-------------|
| <video src="https://github.com/user-attachments/assets/e79d438b-fed4-48b2-b79b-e2c8ff7bb6a1" /> | <video src="https://github.com/user-attachments/assets/090905bb-5f89-4991-883f-a9e1a5f67abc" />

kelson42 commented:
... and if you put a space at start?

kelson42 commented:
> ... and if you put a space at start?

This seems to be the case becauseyou use trim(), but this PR has to introduce a proper test of this (or modify a test to introduce this variant).

MohitMaliFtechiz commented:
> ... and if you put a space at start?

@kelson42 It works in the same manner as spaces at the end. It removes the extra spaces from the written text.


https://github.com/user-attachments/assets/09bd92ad-16d2-4c02-9b65-316ec43eec24

MohitMaliFtechiz commented:
> It is also important to not reissue a suggestion request because the search pattern didn't change.

@kelson42, It would not trigger a suggestion since now we have trimmed the text before getting the result from libkiwix. Recently we had introduced the debouncing in our search functionality where we have added this things there that it only triggers the search when search term is changed.

https://github.com/kiwix/kiwix-android/blob/51727637908400541e5f280c398618df04dbc239/core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/SearchViewModel.kt#L118

MohitMaliFtechiz commented:
Added the UI test case for this scenario.
